### PR TITLE
[Android] Implement the feature to change cache path.

### PR DESCRIPTION
**Pull request by wuhengzhi**
This patch is to add the function to change cache path by adding switch
to command line.
If the user data directory was specified in command line, use this path
instead of the default.

BUG=XWALK-2030

### CODE REVIEW

**Comment by wuhengzhi** [comment]
@lincsoon @axinging @halton Please take a look, thanks.

File: runtime/common/xwalk_switches.cc [hunk 1]
```diff
@@ -32,6 +32,9 @@ const char kXWalkDataPath[] = "data-path";
 #if defined(OS_ANDROID)
 // Specifies the separated folder to save user data on Android.
 const char kXWalkProfileName[] = "profile-name";
+// Specifies the user data directory, which is where the browser will look for
+// all of its state.
+const char kUserDataDir[] = "user-data-dir";
```
Comment (halton) [inline runtime/common/xwalk_switches.cc]: I do not see why this switch is android specific, it should be a general one.

Comment (wuhengzhi) [inline runtime/common/xwalk_switches.cc]: Yes, it is general. In this patch, the implementation is only for android, so I wrap it with OS_ANDROID. I will adjust it.

Comment (halton) [inline runtime/common/xwalk_switches.cc]: I vote do not wrap inside Android logic.

Comment (wuhengzhi) [inline runtime/common/xwalk_switches.cc]: Updated it, thanks.

File: runtime/common/xwalk_switches.h [hunk 2]
```diff
@@ -29,6 +29,8 @@ extern const char kPpapiFlashPath[];
 extern const char kPpapiFlashVersion[];
 #endif
 
+extern const char kUserDataDir[];
```
Comment (lincsoon) [inline runtime/common/xwalk_switches.h]: Isn't this parameter only for Android?

Comment (wuhengzhi) [inline runtime/common/xwalk_switches.h]: From http://peter.sh/experiments/chromium-command-line-switches/, we can see it's a general switch.

File: runtime/browser/xwalk_browser_context.cc [hunk 3]
```diff
@@ -109,11 +109,13 @@ XWalkBrowserContext::CreateZoomLevelDelegate(
...
-  CHECK(PathService::Get(base::DIR_ANDROID_APP_DATA, &result));
   base::CommandLine* cmd_line = base::CommandLine::ForCurrentProcess();
-  if (cmd_line->HasSwitch(switches::kXWalkProfileName))
-    result = result.Append(
-        cmd_line->GetSwitchValuePath(switches::kXWalkProfileName));
+  if (cmd_line->HasSwitch(switches::kUserDataDir))
+    result = cmd_line->GetSwitchValuePath(switches::kUserDataDir);
+  if (result.empty())
+    CHECK(PathService::Get(base::DIR_ANDROID_APP_DATA, &result));
+  result = result.Append(
```
Comment (lincsoon) [inline runtime/browser/xwalk_browser_context.cc]: Don't need to check HasSwitch for `kXWalkProfileName`?

Comment (wuhengzhi) [inline runtime/browser/xwalk_browser_context.cc]: For android, profile name was set as default. This check can be ignored, but the check will increase the robustness, I will add it.

**Comment by lincsoon** [comment]
LGTM

**Comment by axinging** [comment]
LGTM.

**Pull request merged**